src/v/cluster/archival_metadata_stm.cc

```diff
@@ -1491,44 +1491,42 @@ void archival_metadata_stm::apply_reset_scrubbing_metadata() {
     _manifest->reset_scrubbing_metadata();
 }
 
-std::vector<cloud_storage::partition_manifest::lw_segment_meta>
+fragmented_vector<cloud_storage::partition_manifest::lw_segment_meta>
 archival_metadata_stm::get_segments_to_cleanup() const {
     // Include replaced segments to the backlog
     using lw_segment_meta = cloud_storage::partition_manifest::lw_segment_meta;
-    std::vector<lw_segment_meta> backlog = _manifest->lw_replaced_segments();
+    const fragmented_vector<lw_segment_meta> source_backlog
+      = _manifest->lw_replaced_segments();
```

Comment:
Since `lw_replaced_segments` now returns fragmented vector, should we use `const auto` here?

Reply:
https://github.com/redpanda-data/redpanda/pull/16189